internal/implementations/gateway/gateway.go

```diff
@@ -2,41 +2,65 @@ package implementation
 
 import (
 	"github.com/go-logr/logr"
+	"github.com/nginxinc/nginx-kubernetes-gateway/internal/events"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/gateway-api/apis/v1alpha2"
 
 	"github.com/nginxinc/nginx-kubernetes-gateway/internal/config"
 	"github.com/nginxinc/nginx-kubernetes-gateway/pkg/sdk"
 )
 
 type gatewayImplementation struct {
-	conf config.Config
+	conf    config.Config
+	eventCh chan<- interface{}
 }
 
-func NewGatewayImplementation(conf config.Config) sdk.GatewayImpl {
+func NewGatewayImplementation(conf config.Config, eventCh chan<- interface{}) sdk.GatewayImpl {
 	return &gatewayImplementation{
-		conf: conf,
+		conf:    conf,
+		eventCh: eventCh,
 	}
 }
 
 func (impl *gatewayImplementation) Logger() logr.Logger {
 	return impl.conf.Logger
 }
 
-func (impl *gatewayImplementation) ControllerName() string {
-	return impl.conf.GatewayCtlrName
-}
-
 func (impl *gatewayImplementation) Upsert(gw *v1alpha2.Gateway) {
-	if gw.Name == impl.ControllerName() {
-		impl.Logger().Info("Found correct Gateway resource",
+	if gw.Namespace != impl.conf.GatewayNsName.Namespace || gw.Name != impl.conf.GatewayNsName.Name {
+		impl.Logger().Info("Gateway was upserted but ignored based on its namespace/name",
+			"namespace", gw.Namespace,
 			"name", gw.Name,
 		)
 		return
 	}
+
+	impl.Logger().Info("Gateway was upserted",
```

Comment:
I feel like this log is a little misleading. Have we successfully upserted the gateway at this point?

Reply:
yeah, this log just tells that our internal component upserted it, without telling the result of it. The result will be reported as a status of the Gateway. 
If we want to report the status of the Gateway or other resource upsert/delete (which I think can be helpful), I think we need to have a proper approach (feature) for it that will cover all resource types. Also note that since we regenerating the whole graph of resources, after upserting/deleting, we will get valid/invalid information about all resources, not just one.

Reply:
Makes sense.